org-code-javabuilder/lib/src/main/java/org/code/javabuilder/UserInitiatedExceptionKey.java

```diff
@@ -0,0 +1,17 @@
+package org.code.javabuilder;
+
+/** These keys map to client-side keys that are translatable. Examples of what the strings */
+public enum UserInitiatedExceptionKey {
+  // The user's code tried to access a method that it is not allowed to access.
+  illegalMethodAccess,
```

Comment:
Curious for your thoughts here: I'm mapping these directly to our [Java Lab i18n file](https://github.com/code-dot-org/code-dot-org/pull/40507/files#diff-435cc619dfbf34adab6d590b84b07221ea413584c1e89dbfb973b3bf470a7ffb). This streamlines the process of adding more strings so it only happens in the two places. To do that, however, I had to break the convention of one of the two places. i.e. Java enums are all caps, but i18n strings are camelCase. I just picked one without too much consideration. Any reason to prefer one over the other?

Reply:
On the wire, the protocol would look similar to what you've proposed.  An error might look like this:
{
    type: "ServerException"
    value: "InternalCompilerException"
    detail: {
        exceptionMessage: ...
    }
}

The difference is that the values are defined from a server-centric viewpoint.  The "value" is one that makes sense to the server and it will be up to the client to figure out how to display this to the user.